huaweicloud/services/cae/resource_huaweicloud_cae_environment.go

```diff
@@ -0,0 +1,460 @@
+package cae
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+var envResourceNotFoundCodes = []string{"CAE.01500404"}
+
+// @API CAE POST /v1/{project_id}/cae/environments
+// @API CAE GET /v1/{project_id}/cae/jobs/{job_id}
+// @API CAE POST /v1/{project_id}/cae/jobs/{job_id}
+// @API CAE GET /v1/{project_id}/cae/environments
+// @API CAE DELETE /v1/{project_id}/cae/environments/{environment_id}
+func ResourceEnvironment() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceEnvironmentCreate,
+		ReadContext:   resourceEnvironmentRead,
+		UpdateContext: resourceEnvironmentUpdate,
+		DeleteContext: resourceEnvironmentDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(20 * time.Minute),
+			Delete: schema.DefaultTimeout(20 * time.Minute),
```

Comment:
Please add timeouts descriptions in documentation.

Reply:
done.